ceph/utils.py

```diff
@@ -705,46 +705,66 @@ def get_disk_info(node):
     return disks
 
 
-def get_nodes_by_id(cluster, node_names):
+def get_node_by_id(cluster, node_name):
     """
-    Get node object(s) by Node Id
+    Fetch node using provided substring of node name
 
-    Returns all nodes object list if node_names is empty,
-    else node list which matched node ID(eg., 'node1')
+    As per the naming convention used at VM creation, where each node.shortname
+    is framed with hyphen("-") separated string as below,
+        "ceph-<name>-node1-<roles>"
+
+    In this method we use hyphen("-") appended to node_name string to fetch exact node.
```

Comment:
I believe it is sufficient to prefix `-`... but do you think we should also suffix the value?

Reply:
IMO, for now prefix would be fine. 
As we discussed we will try to eliminate the problem by a node attribute in future.

Reply:
Thanks for clarifying that this will be a node attribute in the future. As I was reading this code, I was thinking we could really benefit from storing this as an attribute on the node.